ledger/sandbox/src/test/suite/scala/com/digitalasset/platform/sandbox/stores/ledger/ImplicitPartyAdditionIT.scala

```diff
@@ -0,0 +1,187 @@
+    Ref.QualifiedName(
+      Ref.ModuleName.assertFromString("moduleName"),
+      Ref.DottedName.assertFromString("name")
+    )
+  )
+
+  private def textValue(t: String) =
+    VersionedValue(ValueVersions.acceptedVersions.head, ValueText(t))
+
+  /** Overriding this provides an easy way to narrow down testing to a single implementation. */
+  override protected def fixtureIdsEnabled: Set[BackendType] =
+    Set(BackendType.InMemory, BackendType.Postgres)
+
+  override protected def constructResource(index: Int, fixtureId: BackendType): Resource[Ledger] =
+    fixtureId match {
+      case BackendType.InMemory =>
+        LedgerResource.inMemory(ledgerId, timeProvider)
+      case BackendType.Postgres =>
+        LedgerResource.postgres(ledgerId, timeProvider)
+    }
+
+  private def publishSingleNodeTx(
+      ledger: Ledger,
+      submitter: String,
+      commandId: String,
+      node: GenNode[NodeId, TContractId, Value[TContractId]]): Future[SubmissionResult] = {
+    val event1: NodeId = NodeId.unsafeFromIndex(0)
+
+    val transaction = GenTransaction[NodeId, TContractId, Value[TContractId]](
+      Map(event1 -> node),
+      ImmArray(event1),
+      Set.empty
+    )
+
+    val submitterInfo = SubmitterInfo(
+      Ref.Party.assertFromString(submitter),
+      Ref.LedgerString.assertFromString("appId"),
+      Ref.LedgerString.assertFromString(commandId),
+      Timestamp.assertFromInstant(MRT)
+    )
+
+    val transactionMeta = TransactionMeta(
+      Timestamp.assertFromInstant(LET),
+      Some(Ref.LedgerString.assertFromString("wfid"))
+    )
+
+    ledger.publishTransaction(submitterInfo, transactionMeta, transaction)
+  }
+
+  val LET = Instant.EPOCH.plusSeconds(10)
+  val MRT = Instant.EPOCH.plusSeconds(10)
+
+  "A Ledger" should {
+    "implicitly add parties mentioned in a transaction" in allFixtures { ledger =>
+      for {
+        createResult <- publishSingleNodeTx(
+          ledger,
+          "create-signatory",
+          "CmdId1",
+          NodeCreate(
+            AbsoluteContractId("cId1"),
+            ContractInst(
+              templateId1,
+              textValue("some text"),
+              "agreement"
+            ),
+            None,
+            Set("create-signatory"),
+            Set("create-stakeholder"),
+            Some(KeyWithMaintainers(textValue("some text"), Set("create-signatory")))
+          )
+        )
+        exerciseResult <- publishSingleNodeTx(
+          ledger,
+          "exercise-signatory",
+          "CmdId2",
+          NodeExercises(
+            AbsoluteContractId("cId1"),
+            templateId1,
+            Ref.ChoiceName.assertFromString("choice"),
+            None,
+            true,
+            Set("exercise-signatory"),
+            textValue("choice value"),
+            Set("exercise-stakeholder"),
+            Set("exercise-signatory"),
+            Set("exercise-signatory"),
+            ImmArray.empty,
+            None
+          )
+        )
```

Comment:
Can you add cases for the other `GenNode` types as well?

Reply:
Added a test case for `NodeFetch`. `NodeLookupByKey` should not (and does not, in the current implementation) add any new parties.